x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/OperatorTestCase.java

```diff
@@ -148,6 +149,47 @@ public final void testSimpleWithCranky() {
         assertThat(inputFactoryContext.breaker().getUsed(), equalTo(0L));
     }
 
+    public final void testSimpleDoesNotLeak() {
+        DriverContext inputFactoryContext = driverContext();
+        List<Page> input = CannedSourceOperator.collectPages(simpleInput(inputFactoryContext.blockFactory(), between(1_000, 10_000)));
+
+        List<Block> inputBlocks = new ArrayList<>();
+        for (Page p : input) {
+            for (int i = 0; i < p.getBlockCount(); i++) {
+                inputBlocks.add(p.getBlock(i));
+            }
+        }
+
+        BlockFactory nonBreakingBlockFactory = BlockFactory.getNonBreakingInstance();
+        BigArrays bigArrays = new MockBigArrays(
+            PageCacheRecycler.NON_RECYCLING_INSTANCE,
+            nonBreakingBlockFactory.bigArrays().breakerService()
+        ).withCircuitBreaking();
+        DriverContext driverContext = new DriverContext(bigArrays, nonBreakingBlockFactory);
+
+        Operator operator = simple(bigArrays).get(driverContext);
+        List<Page> result = drive(operator, input.iterator(), driverContext);
+
+        List<Block> resultBlocks = new ArrayList<>();
+        for (Page p : result) {
+            for (int i = 0; i < p.getBlockCount(); i++) {
+                resultBlocks.add(p.getBlock(i));
+            }
+        }
+
+        for (Page p : result) {
```

Comment:
I wonder if we can merge this into the `assertSimple`. I mean, if use the MockBlockFactory and check the breaker I think it'll already catch this as is. Those just aren't using that block factory.

Reply:
++

Reply:
Yep, works just fine. Done it this way now, this also increases the parametrization nicely as `assertSimple` is run for both small and large inputs.